Engine/DataFeeds/LiveOptionChainProvider.cs

```diff
@@ -42,18 +62,158 @@ static LiveOptionChainProvider()
         }
 
         /// <summary>
-        /// Gets the list of option contracts for a given underlying symbol
+        /// Gets the option chain associated with the underlying Symbol
         /// </summary>
-        /// <param name="symbol">The underlying symbol</param>
-        /// <param name="date">The date for which to request the option chain (only used in backtesting)</param>
-        /// <returns>The list of option contracts</returns>
-        public IEnumerable<Symbol> GetOptionContractList(Symbol symbol, DateTime date)
+        /// <param name="underlyingSymbol">Underlying symbol to get the option chain for</param>
+        /// <param name="date">Unused</param>
+        /// <returns>Option chain</returns>
+        /// <exception cref="ArgumentException">Option underlying Symbol is not Future or Equity</exception>
+        public IEnumerable<Symbol> GetOptionContractList(Symbol underlyingSymbol, DateTime date)
         {
-            if (symbol.SecurityType != SecurityType.Equity)
+            if (underlyingSymbol.SecurityType == SecurityType.Equity)
+            {
+                // Source data from TheOCC if we're trading equity options
+                return GetEquityOptionContractList(underlyingSymbol, date);
+            }
+            if (underlyingSymbol.SecurityType == SecurityType.Future)
             {
-                throw new NotSupportedException($"LiveOptionChainProvider.GetOptionContractList(): SecurityType.Equity is expected but was {symbol.SecurityType}");
+                // We get our data from CME if we're trading future options
+                return GetFutureOptionContractList(underlyingSymbol, date);
             }
 
+            throw new ArgumentException("Option's Underlying SecurityType is not supported. Supported types are: Equity, Future");
+        }
+
+        private IEnumerable<Symbol> GetFutureOptionContractList(Symbol futureContractSymbol, DateTime date)
+        {
+            var symbols = new List<Symbol>();
+            var retries = 0;
+            var maxRetries = 5;
+
+            while (++retries <= maxRetries)
+            {
+                try
+                {
+                    _rateGate.WaitToProceed();
+
+                    var productResponse = _client.GetAsync(CMEProductSlateURL.Replace(CMESymbolReplace, futureContractSymbol.ID.Symbol))
+                        .SynchronouslyAwaitTaskResult();
+
+                    productResponse.EnsureSuccessStatusCode();
```

Comment:
Cool! Believe `productResponse` should be disposed of, same for the different responses bellow